### ISSUE TITLE
Register URL Protocol with OS

### ISSUE THREAD

**Issue by AlbinoGeek** [fyne-io/fyne#1847]
### Is your feature request related to a problem? Please describe:

Now that we have the URI rework...

I want `matrix://` scheme URIs to open my installed `fyne` application.

### Is it possible to construct a solution with the existing API?

Sort of. This is a change to `fyne package` .

### Describe the solution you'd like to see:

Open for suggestions, I would love to control this from code, but on some platforms this can only be done at install-time.

### References

Mac OS X: https://gist.github.com/nathankerr/38d8b0d45590741b57f5f79be336f07c

Linux: https://unix.stackexchange.com/a/497147 (Desktop Entry with `MimeType` and `xdg`)

Windows: https://docs.microsoft.com/en-us/windows/win32/search/-search-3x-wds-ph-install-registration

**Comment by Jacalz** [comment]
I agree. This would be very cool. Support for opening files with a specific file extension/mimetype probably more or less falls in the same ballpark (at least on Linux) 👍

**Comment by stuartmscott** [comment]
Also Android: https://developer.android.com/training/app-links/

**Comment by gedw99** [comment]
This is also what I need too :) 

I expect that it would be developed as part of the Packaging code.

So then from Finder / Explorer when you right click a file type you App us one of the options to edit that file. 

Anyone know if this is Roadmapped yet or if anyone has taken this task ?

**Comment by gedw99** [comment]
@AlbinoGeek the example code links look very good. Thank you for adding these !!!

mid you want we can work on this together ??
